Java/core/src/main/java/com/amazon/randomcutforest/state/store/SmallNodeStoreMapper.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.state.store;
+
+import static com.amazon.randomcutforest.CommonUtils.checkState;
+import static com.amazon.randomcutforest.tree.AbstractCompactRandomCutTree.NULL;
+
+import java.util.Arrays;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import com.amazon.randomcutforest.CommonUtils;
+import com.amazon.randomcutforest.config.Precision;
+import com.amazon.randomcutforest.state.IStateMapper;
+import com.amazon.randomcutforest.store.SmallNodeStore;
+import com.amazon.randomcutforest.util.ArrayPacking;
+
+@Getter
+@Setter
+public class SmallNodeStoreMapper implements IStateMapper<SmallNodeStore, NodeStoreState> {
+
+    /**
+     * if single precision, then stores the cut information as a float array
+     * (converted to bytes)
+     */
+    private Precision precision = Precision.FLOAT_64;
```

Comment:
Could you explain why precision is hardcoded?

Reply:
Um ... :) This is leftover from the default being FLOAT_64 and was explicitly overridden for FLOAT_32 (in NodeStore). So it should be changed in NodeStoreMapper. For SmallNodeStore this should be FLOAT_32. This is needed to store the cuts accurately because they are written out into the byte[]. Cleaning up.